From f0cb90f853fb4b7e80f5fb64caf1339ed05cc9c4 Mon Sep 17 00:00:00 2001
From: Roman Shaposhnik <rvs@cloudera.com>
Date: Mon, 8 Oct 2012 14:23:52 -0700
Subject: [PATCH 150/795] MR1: CLOUDERA-BUILD. mr1 tarball build requires native (CDH-7918)
 (cherry picked from commit 982f4df0e3cbd63dde30e2c547628e5b51cfae91)

(cherry picked from commit b0daba2471ffa15e25ecf3f5c05588b010a5ce89)
---
 build.xml |   27 ++++++++++++++++++---------
 1 files changed, 18 insertions(+), 9 deletions(-)

diff --git a/build.xml b/build.xml
index ceff65a..882dfb6 100644
--- a/build.xml
+++ b/build.xml
@@ -1271,12 +1271,16 @@
     <chmod perm="ugo+x" type="file">
         <fileset dir="${dist.dir}/src/c++/pipes/debug"/>
     </chmod>
-    <chmod perm="ugo+x" type="file">
-        <fileset dir="${dist.dir}/c++/">
-          <include name="*/bin/*" />
-        </fileset>
-    </chmod>
-
+    <if>
+      <istrue value="${compile.c++}"/>
+      <then>
+        <chmod perm="ugo+x" type="file">
+            <fileset dir="${dist.dir}/c++/">
+              <include name="*/bin/*" />
+            </fileset>
+        </chmod>
+      </then>
+    </if>
   </target>
 
   <!-- ================================================================== -->
@@ -1398,9 +1402,14 @@
       <fileset dir="cloudera" />
     </copy>
 
-    <copy todir="${dist.dir}/c++" includeEmptyDirs="false">
-      <fileset dir="${build.dir}/c++-examples"/>
-    </copy>
+    <if>
+      <istrue value="${compile.c++}"/>
+      <then>
+        <copy todir="${dist.dir}/c++" includeEmptyDirs="false">
+          <fileset dir="${build.dir}/c++-examples"/>
+        </copy>
+      </then>
+    </if>
     <!-- END CLOUDERA HOOKS -->
 
     <copy todir="${dist.dir}/.eclipse.templates">
-- 
1.7.0.4

